en/scripting/modules/example.md

```diff
@@ -0,0 +1,248 @@
+
+```ts
+import proto from './Proto.js/proto.js';
+console.log(proto.pkg1.Bar);
+```
+
+Here it is still important to state that if anyone gets an error when importing, indicating that proto is not exported by default, there are two solutions:
+
+- It can be solved by allowing default import fields for modules containing default exports in `tsconfig.json`
+
+    ```json
+    "compilerOptions": {
+      "allowSyntheticDefaultImports": true,
+    }
+    ```
+
+- Add the default export
+
+    Create a Tools/wrap-pbts-result.js file in the project directory with the following script code:
+
+    ```js
+    const fs = require('fs');
+    const ps = require('path');
+    const file = ps.join(__dirname, '...' , 'proto.js', 'proto.d.ts');
+    const original = fs.readFileSync(file, { encoding: 'utf8' });
+    fs.writeFileSync(file, `
+    namespace proto {
+        ${original}
+    }
+    export default proto;
+    `);
+    ```
+
+    Replace the original ``build-proto:pbts`` command with the following:
+
+    ```json
+    "build-proto:pbts": "pbts --main --out ./Proto.js/proto.d.ts ./Proto.js/*.js && node ./Tools/wrap-pbts-result.js"
+    ```
+
+Eventually, it will be ready to run directly. For the full project content, please refer to: [npm-case](https://github.com/cocos-creator/example-3d/tree/master/npm-case).
+
+> **Note**: the packaged js file can be placed in the `assets` directory of the project or elsewhere in the project. The js file in the `assets` directory no longer needs to be checked for export as a plugin, please be aware of that.
+
+### Case 3: use of lodash-es
+
+Similar to [Case 1: protobufjs use](# Case 1: protobufjs-use), install the `lodash-es` package. We know that the entry file is `lodash.js`, and the entry file also automatically helps to export all submodules under it in **ESM** module format, and according to `type` it also confirms that it is currently an **ESM** module. Therefore, it is possible to import any module directly. Let's take the `test.ts` script asset under `assets` as an example, and introduce the submodules within `lodash`.
+
+```ts
+import { array, add } from 'lodash-es';
+```
+
+At this point, you will find that an error is reported at the code level, but it actually works. This is because there is a clear distinction between JavaScript, which is dynamically typed, and TypeScript, which is statically typed, so there is no way to know the exact type of the exported module when using js scripts, and the best thing to do is to declare a type definition file .d.ts. Fortunately, but when we hover over the error, there is a hint The type declaration file for the `lodash` module can be installed by running `npm i --save-dev @types/lodash-es`. After installation, restart VS Code and you will see that the error message disappears, along with the code hint.
```

Comment:
```suggestion
At this point, notice that an error is reported at the code level, but it actually works. This is because there is a clear distinction between JavaScript, which is dynamically typed, and TypeScript, which is statically typed, this means there is no way to know the exact type of the exported module when using JavaScript, and the best thing to do is to declare a type definition file `.d.ts`. Fortunately, but when hovering over the error, there is a hint `The type declaration file for the ``lodash`` module can be installed by running ``npm i --save-dev @types/lodash-es```. After installation, restart VS Code and notice the error message disappears, along with the code hint.
```